staging/src/k8s.io/client-go/tools/clientcmd/api/helpers.go

```diff
@@ -189,3 +191,76 @@ func MakeAbs(path, base string) (string, error) {
 	}
 	return filepath.Join(base, path), nil
 }
+
+// RedactSecrets replaces any sensitive values with REDACTED
+func RedactSecrets(config *Config) error {
```

Comment:
Returning a copy that is redacted instead of mutating the original seems like more obvious behavior. With generated deepcopy methods, this ought not be too bad to do.

Reply:
I'd definitely support that rewrite in a followup PR. This will be a change in behavior of all the public functions.

Reply:
+1 for what David says. But yeah, this package likes in-place updates.